app/master/build.py

```diff
@@ -58,6 +60,10 @@ def __init__(self, build_request):
         self._postbuild_tasks_are_finished = False
         self._timing_file_path = None
 
+        self._state_timestamps = {status.lower(): None  # initialize all timestamps to None
```

Comment:
Really this only matters for the api response -- I wanted those field names to be lowercase. But yeah, the keys in the dict don't have to be lowercased. I'll see if it would be simpler just to lowercase for the api response.